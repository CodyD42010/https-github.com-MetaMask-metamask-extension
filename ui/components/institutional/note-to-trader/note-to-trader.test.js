import { render } from '@testing-library/react';
import React from 'react';
import sinon from 'sinon';
import NoteToTrader from './note-to-trader';

describe('NoteToTrader', () => {
  it('should render the Note to trader component', () => {
    const props = {
      placeholder: '',
      maxLength: '280',
      noteText: 'some text',
      labelText: 'Transaction note',
      onChange: sinon.spy(),
    };

    const { getByTestId, container } = render(<NoteToTrader {...props} />);

    expect(getByTestId('transaction-note')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
