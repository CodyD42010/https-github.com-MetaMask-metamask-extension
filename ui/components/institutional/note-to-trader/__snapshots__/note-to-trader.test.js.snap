// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteToTrader should render the Note to trader component 1`] = `
<div>
  <div
    class="box confirm-page-container-content__data box--flex-direction-row"
  >
    <div
      class="box box--display-flex box--flex-direction-row"
    >
      <div
        class="box note-header box--display-flex box--flex-direction-row box--justify-content-space-between"
      >
        <label
          class="box mm-text mm-label mm-label--html-for mm-text--body-md mm-text--font-weight-bold box--display-inline-flex box--flex-direction-row box--align-items-center box--color-text-default"
          for="transaction-note"
        >
          Transaction note
        </label>
        <p
          class="box mm-text note-header__counter mm-text--body-md box--flex-direction-row box--color-text-default"
        >
          9
          /
          280
        </p>
      </div>
      <div
        class="box note-field box--display-flex box--flex-direction-column"
      >
        <textarea
          data-testid="transaction-note"
          id="transaction-note"
          maxlength="280"
          placeholder=""
        >
          some text
        </textarea>
      </div>
    </div>
  </div>
</div>
`;
